crates/cosmos/cosmos-client-components/src/impls/queries/ack_packets.rs

```diff
@@ -0,0 +1,43 @@
+use cgp_core::prelude::*;
+use futures::stream::{self, StreamExt, TryStreamExt};
+use hermes_relayer_components::chain::traits::queries::ack_packets::AckPacketsQuerier;
+use hermes_relayer_components::chain::traits::queries::ack_packets::CanQueryAckPacket;
+use hermes_relayer_components::chain::traits::types::ibc::HasIbcChainTypes;
+use hermes_relayer_components::chain::traits::types::ibc_events::write_ack::HasWriteAckEvent;
+use hermes_relayer_components::chain::types::aliases::WriteAckEventOf;
+
+pub struct QueryAckPacketsConcurrently;
+
+#[async_trait]
+impl<Chain, Counterparty> AckPacketsQuerier<Chain, Counterparty> for QueryAckPacketsConcurrently
+where
+    Chain: CanQueryAckPacket<Counterparty>,
+    Counterparty: HasIbcChainTypes<Chain> + HasWriteAckEvent<Chain>,
+{
+    async fn query_ack_packets_from_sequences(
+        chain: &Chain,
+        channel_id: &Counterparty::ChannelId,
+        port_id: &Counterparty::PortId,
+        counterparty_channel_id: &Chain::ChannelId,
+        counterparty_port_id: &Chain::PortId,
+        sequences: &[Chain::Sequence],
+        height: &Chain::Height,
+    ) -> Result<Vec<(Chain::OutgoingPacket, WriteAckEventOf<Chain, Counterparty>)>, Chain::Error>
+    {
+        let ack_packets = stream::iter(sequences)
+            .then(|sequence| {
```

Comment:
I think you might need [`for_each_concurrent`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.for_each_concurrent) to run the stream concurrently.

Reply:
Should we also update the `send_packets.rs` with the concurrent call? https://github.com/informalsystems/hermes-sdk/blob/59ac2f025522dd021efee440c6639173f518bfc8/crates/cosmos/cosmos-client-components/src/impls/queries/send_packets.rs#L25

Reply:
Sounds good.